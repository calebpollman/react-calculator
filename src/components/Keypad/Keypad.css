.keypad-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    height: var(--keypad-height);
    padding: 2%;
    width: var(--keypad-width);
}

.numbers-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    height: 80%;
    width: 75%;
}

.operators-container {
    display: flex;
    flex-direction: column;
    height: 80%;
    width: 25%;
}

.submit-container {
    height: 20%;
    width: 100%;
}