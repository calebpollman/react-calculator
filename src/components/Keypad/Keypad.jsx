import React from 'react';
import PropTypes from 'prop-types';
import './Keypad.css';
import Key from '../Key/Key';

const Keypad = ({ callOperator, numbers, operators, setOperator, updateDisplay }) =>
{
    const numberKeys = numbers.map(number => (
        <Key
            key={number}
            keyAction={updateDisplay}
            keyType="number-key"
            keyValue={number}
        />)
    );
    const operatorKeys = operators.map(operator => (
        <Key
            key={operator}
            keyAction={setOperator}
            keyType="operator-key"
            keyValue={operator}
        />)
    );

    return (
        <div className="keypad-container">
            <div className="numbers-container">
                {numberKeys}
            </div>
            <div className="operators-container">
                {operatorKeys}
            </div>
            <div className="submit-container">
                <Key
                    keyAction={callOperator}
                    keyType="submit-key"
                    keyValue="="
                />
            </div>
        </div>
    );
}

Keypad.propTypes = {
    callOperator: PropTypes.func.isRequired,
    numbers: PropTypes.array.isRequired,
    operators: PropTypes.array.isRequired,
    setOperator: PropTypes.func.isRequired,
    updateDisplay: PropTypes.func.isRequired,
}

export default Keypad;