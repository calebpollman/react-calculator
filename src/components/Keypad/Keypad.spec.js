import React from 'react';
import { shallow, mount } from 'enzyme';
import Keypad from './Keypad';

describe('Keypad', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Keypad
                callOperator={jest.fn()}
                numbers={[]}
                operators={[]}
                setOperator={jest.fn()}
                updateDisplay={jest.fn()}
            />
        );
    });

    it('should render 4 <div />', () => {
        expect(wrapper.find('div').length).toEqual(4);
    });

    it('should render an instance of the Key component for each index of numbers, operators, and the submit Key', () => {
        const numbers = ['0', '1'];
        const operators = ['+', '-'];
        const submit = 1;
        const keyTotal = numbers.length + operators.length + submit;
        wrapper.setProps({ numbers, operators });
        expect(wrapper.find('Key').length).toEqual(keyTotal);
    });
});

describe('MountedKeypad', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <Keypad
                callOperator={jest.fn()}
                numbers={[]}
                operators={[]}
                setOperator={jest.fn()}
                updateDisplay={jest.fn()}
            />
        )});

    it('renders the values of numbers', () => {
        wrapper.setProps({numbers: ['0', '1', '2']});
        expect(wrapper.find('.numbers-container').text()).toEqual('012');
    });

    it('renders the values of operators', () => {
        wrapper.setProps({operators: ['+', '-', '*', '/']});
        expect(wrapper.find('.operators-container').text()).toEqual('+-*รท');
    });

    it('should render an instance of the Key component', () => {
        expect(wrapper.find('Key').length).toEqual(1);
    });

    it('should render an instance of the Key component for each index of numbers, operators, and the submit Key', () => {
        const numbers = ['0', '1'];
        const operators = ['+', '-'];
        const submit = 1;
        const keyTotal = numbers.length + operators.length + submit;
        wrapper.setProps({ numbers, operators });
        expect(wrapper.find('Key').length).toEqual(keyTotal);
    });

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

});