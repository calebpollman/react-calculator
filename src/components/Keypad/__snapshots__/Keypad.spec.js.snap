// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MountedKeypad should render correctly 1`] = `ReactWrapper {}`;
